.github/workflows/main.yaml

```diff
@@ -0,0 +1,102 @@
+name: main
+
+env:
+  AWS_REGION: us-east-1
+  AWS_ROLE_DURATION: 3600
+  # role generated from https://github.com/Sceptre/sceptre-aws/blob/master/config/prod/gh-oidc-sceptre-tests.yaml
+  AWS_ROLE: TBD
+
+on:
+  push:
+    branches:
+      - '*'
+    tags:
+      - '^v([0-9]+)\.([0-9]+)\.([0-9]+)(?:([0-9A-Za-z-]+(?:\.[0-9A-Za-z-]+)*))?(?:\+[0-9A-Za-z-]+)?$'
+  pull_request:
+    branches:
+      - '*'
+
+jobs:
+  linting:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - name: run pre-commit
+        run: poetry run pre-commit run --all-files --show-diff-on-failure
+  packaging:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - name: build package
+        run: poetry build
+  unit-tests:
+    needs: linting
+    strategy:
+      fail-fast: true
+      matrix:
+        os: [ "ubuntu-latest" ]
+        python-version: [ "3.8", "3.9", "3.10", "3.11", "3.12" ]
+    runs-on: ${{ matrix.os }}
+    steps:
+      - name: Check out repository
+        uses: actions/checkout@v4
+      - name: Set up python ${{ matrix.version }}
+        id: setup-python
+        uses: actions/setup-python@v5
+        with:
+          python-version: ${{ matrix.version }}
+          cache: 'poetry'
+      - name: Install dependencies
+        run: poetry install --all-extras -v
+      - name: run tests
+        run: poetry run tox -e py
+          poetry run tox -e report
+  docker-build:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Build Docker Image
+        uses: docker/build-push-action@v5
+        with:
+          context: .
+  integration-tests:
+    if: ${{ github.event_name != 'pull_request' }}
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      # Update poetry for https://github.com/python-poetry/poetry/issues/7184
+      - name: update poetry
+        run: poetry self update --no-ansi
+      - name: Assume AWS role
+        uses: aws-actions/configure-aws-credentials@v2
+        with:
+          aws-region: ${{ env.AWS_REGION }}
+          role-to-assume: ${{ env.AWS_ROLE }}
+          role-session-name: GHA-${{ github.repository_owner }}-${{ github.event.repository.name }}-${{ github.run_id }}
+          role-duration-seconds: ${{ env.AWS_ROLE_DURATION }}
+      - name: run tests
+        run: poetry run behave --junit --junit-directory build/behave
```

Comment:
on circleci we [split integration tests using the circle ci runner](https://circleci.com/docs/use-the-circleci-cli-to-split-tests/) so that they run in parallel on multiple instances.  I'm not sure how to replicate that in GH actions but i think it will be necessary because these tests take along time to complete.  In parallel mode, they were taking 30mins to complete on circleci.

Reply:
It's probably worth testing to see how long they take on the GitHub runners. Probably not much quicker, if at all, but I wouldn't prematurely optimize.

That said, if I were attempting this I would do two things:

1. Break this out into a separate workflow -- this part isn't absolutely necessary, but will improve readability. And means you don't need the `if:` at the top.
2. Use a matrix build, where the matrix variables are test names. Then your command here would add `--include ${{ matrix.tests }}`.

Also, `behave` has a `--parallel` option. Could that help out here?

Reply:
> It's probably worth testing to see how long they take on the GitHub runners. Probably not much quicker, if at all, but I wouldn't prematurely optimize.

Good idea.  Let's do the simple and, if needed, optimize on a follow on PR.  I'm not sure running in separate workflows or as a matrix would work though because there are lots of dependencies between cloudformation stacks in those tests.

> Also, behave has a --parallel option. Could that help out here?

[Behave issue #235](https://github.com/behave/behave/issues/235#issuecomment-426567126) indicates that it's there but it doesn't work.  

```
While the behave that is present in behave-parallel, doesnt have that support. And behave 1.2.5 doesnt have "processes" parameter support (https://github.com/hugeinc/behave-parallel/blob/upstreamsync/behave/configuration.py#L71)
```

Also I tested that as an option when setting up split tests in circleci and it didn't really help at all.